starknet_py/utils/merkle_tree.py

```diff
@@ -0,0 +1,52 @@
+from dataclasses import dataclass, field
+from typing import List, Tuple
+
+from starknet_py.hash.hash_method import HashMethod
+
+
+@dataclass
+class MerkleTree:
+    """
+    Dataclass representing a MerkleTree object.
+    """
+
+    leaves: List[int]
+    hash_method: HashMethod
+    root_hash: int = field(init=False)
+    levels: List[List[int]] = field(init=False)
+
+    def __post_init__(self):
+        if self.leaves:
+            self.root_hash, self.branches = MerkleTree.build(
+                self.leaves, self.hash_method
+            )
+
+    @staticmethod
+    def build(
+        leaves: List[int], hash_method: HashMethod
+    ) -> Tuple[int, List[List[int]]]:
+        if not leaves:
+            raise ValueError("Cannot build Merkle tree from an empty list of leaves.")
+
+        curr_level_nodes = leaves[:]
+        branches: List[List[int]] = []
+
+        while len(curr_level_nodes) > 1:
+            if len(curr_level_nodes) != len(leaves):
+                branches.append(curr_level_nodes[:])
+
+            new_nodes = []
+            for i in range(0, len(curr_level_nodes), 2):
+                a, b = (
+                    curr_level_nodes[i],
+                    curr_level_nodes[i + 1] if i + 1 < len(curr_level_nodes) else 0,
+                )
+                new_nodes.append(MerkleTree.hash(a, b, hash_method))
+
+            curr_level_nodes = new_nodes
+
+        return curr_level_nodes[0], branches
```

Comment:
Why don't we store leaves and root hash in the `levels`?

Reply:
I've added it in these two commits
https://github.com/software-mansion/starknet.py/pull/1363/commits/c490faf860c77bd7f060f087987076c13e37eeb6
https://github.com/software-mansion/starknet.py/pull/1363/commits/faee6d4b97fa6e6772978090bb683611f2c84636
Now we return leaves, levels and the root hash.